app/src/main/java/com/github/se/gomeet/ui/mainscreens/events/MyEventInfo.kt

```diff
@@ -266,6 +292,37 @@ fun MyEventInfo(
   if (showShareEventDialog) {
     ShareDialog("Event", eventId, onDismiss = { showShareEventDialog = false })
   }
+
+  if (showDeleteEventDialog) {
+    DeleteEventDialog(
+        onConfirm = {
+          coroutineScope.launch {
+            myEvent.value!!.participants.forEach { participant ->
+              userViewModel.gotKickedFromEvent(eventId, participant)
+
+              if (participant == currentUser.value!!.uid) {
+                userViewModel.userDeletesEvent(eventId, participant)
+              }
+            }
+
+            myEvent.value!!.pendingParticipants.forEach { pendingParticipant ->
+              userViewModel.invitationCanceled(eventId, pendingParticipant)
+            }
+
+            val allUsers = userViewModel.getAllUsers()
+
+            allUsers!!.forEach { user ->
+              if (user.myFavorites.contains(eventId)) {
+                userViewModel.removeFavoriteEvent(eventId, user.uid)
+              }
+            }
+
+            eventViewModel.removeEvent(eventId)
```

Comment:
This viewmodel function should incorporate the whole logic of this lambda

Reply:
Yes but to have the whole logic in here, the `EventViewModel` would have needed a `UserViewModel` and I wasn't sure if it was good practise to call another viewModel inside of a viewModel when implementing the needed functions in `EventViewModel`.

Reply:
I suppose that makes sense. I don't know if what I'm about to suggest is good practice or not, but what I would've done is put the function in EventViewModel and then made it communicate directly with another static UserRepository function, rather than going through the UserViewModel from the EventViewModel. But then again, it might be bad practice to access the UserRepository from the EventViewModel. Object-oriented is confusing sometimes ngl